mlflow/utils/_capture_transformers_modules.py

```diff
@@ -0,0 +1,83 @@
+"""
+This script should be executed in a fresh python interpreter process using `subprocess`.
+"""
+import json
+import os
+import sys
+
+import mlflow
+from mlflow.exceptions import MlflowException
+from mlflow.utils._capture_modules import (
+    _CaptureImportedModules,
+    parse_args,
+    store_imported_modules,
+)
+
+
+class _CaptureImportedModulesForHF(_CaptureImportedModules):
+    """
+    A context manager to capture imported modules by temporarily applying a patch to
+    `builtins.__import__` and `importlib.import_module`.
+    Used for 'transformers' flavor only.
+    """
+
+    def __init__(self, module_to_throw):
+        super().__init__()
+        self.module_to_throw = module_to_throw
+
+    def _record_imported_module(self, full_module_name):
+        if full_module_name == self.module_to_throw or full_module_name.startswith(
+            f"{self.module_to_throw}."
+        ):
+            raise ImportError(f"Disabled package {full_module_name}")
+        return super()._record_imported_module(full_module_name)
+
+    def __enter__(self):
+        # Patch the environment variables to disable module_to_throw
+        # https://github.com/huggingface/transformers/blob/3658488ff77ff8d45101293e749263acf437f4d5/src/transformers/utils/import_utils.py#L60-L62
+        self.use_tf = os.environ.get("USE_TF")
+        self.use_torch = os.environ.get("USE_TORCH")
+        if self.module_to_throw == "tensorflow":
+            os.environ["USE_TORCH"] = "TRUE"
+        elif self.module_to_throw == "torch":
+            os.environ["USE_TF"] = "TRUE"
+
+        return super().__enter__()
+
+    def _restore_env_var(self, key, value):
+        if key in os.environ:
+            if value is None:
+                del os.environ[key]
+            else:
+                os.environ[key] = value
```

Comment:
Nit: Because we run `_CaptureImportedModulesForHF` in a child process, I think we don't need to restore env vars and it can simplify the code.  CC @harupy to double check.

Reply:
@serena-ruan +1, but not a blocker.

Reply:
Sounds good!